src/Data/Functor/Contravariant/Decide.hs

```diff
@@ -89,13 +92,31 @@ class Contravariant f => Decide f where
     -- returns the wrapped/combined consumer.
     decide :: (a -> Either b c) -> f b -> f c -> f a
 
+-- | Generic 'decide'. Caveats:
+--
+--   1. Will not compile if @f@ is a sum type.
+--   2. Will not compile if @f@ contains fields that do not mention its type variable.
+--
+-- @since 5.3.8
+gdecide :: (Generic1 f, Decide (Rep1 f)) => (a -> Either b c) -> f b -> f c -> f a
```

Comment:
Another caveat for `gdecide` (as well as every other method in the `Data.Functor.Contravariant.*` space) is that it won't be able to support data types where a field type mentions `a` in a contravariant position, e.g.,

```
> data Bar a = MkBar (a -> Int) deriving Generic1; instance Decide Bar where decide = gdecide

<interactive>:20:40: error:
    * Can't make a derived instance of `Generic1 Bar':
        Constructor `MkBar' applies a type to an argument involving the last parameter
                            but the applied type is not of kind * -> *
    * In the data declaration for `Bar'
```

This is a limitation of `Generic1` moreso than `gdecide` itself, but I think it's worth pointing out, since that is the most common use case for these `Contravariant`-inspired classes.